src/concrete/ml/common/utils.py

```diff
@@ -583,3 +582,33 @@ def all_values_are_of_dtype(*values: Any, dtypes: Union[str, List[str]]) -> bool
         supported_dtypes[dtype] = supported_dtype
 
     return all(_is_of_dtype(value, supported_dtypes) for value in values)
+
+
+def check_isclose_and_shape(a, b, **kwargs):
+    """Check if two numpy arrays are element-wise equal within a tolerances and have the same shape.
+
+    Args:
+        a (numpy.ndarray): The first input array
+        b (numpy.ndarray): The second input array
+        **kwargs: Additional keyword arguments to be passed to `numpy.isclose`.
+
+    Returns:
+        bool: True if the arrays have the same shape and all elements are equal within the specified
+            tolerances, False otherwise.
+    """
+    return a.shape == b.shape and numpy.isclose(a, b, **kwargs).all()
+
+
+def check_allclose_and_shape(a, b, **kwargs):
+    """Check if two numpy arrays are equal within a tolerances and have the same shape.
+
+    Args:
+        a (numpy.ndarray): The first input array
+        b (numpy.ndarray): The second input array
+        **kwargs: Additional keyword arguments to be passed to `numpy.isclose`.
+
+    Returns:
+        bool: True if the arrays have the same shape and all elements are equal within the specified
+            tolerances, False otherwise.
+    """
+    return a.shape == b.shape and numpy.allclose(a, b, **kwargs)
```

Comment:
maybe check that a and b are numpy arrays as well first, else we might have an undesired error where accessing shapes !